### PR TITLE
Initial Azure destination

### PR DESCRIPTION
**Pull request by jsantos17**
[ch9646]

### CODE REVIEW

**Comment by jsantos17** [comment]
Depends on slamdata/async-blobstore#34

🛑 up because of `async-blobstore` snapshot

**Comment by jsantos17** [comment]
Released as `quasar-destination-azure_2.12-0.0.1-02e6578`.

File: core/src/test/scala/quasar/destination/azure/AzureConfigSpec.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        "accountName" := "some-name",
+        "accountKey" := "some-key"))
+
+    testConfig.as[AzureConfig].result must beRight(AzureConfig(
+      ContainerName("some-name"),
+      StorageUrl("https://some-name.blob.core.windows.net"),
+      AzureCredentials(
+        AccountName("some-name"),
+        AccountKey("some-key"))))
+  }
```
Comment (rintcius) [inline core/src/test/scala/quasar/destination/azure/AzureConfigSpec.scala]: Maybe add some more tests? E.g. credentials are mandatory & that credentials are being sanitized ok.

Comment (jsantos17) [reply core/src/test/scala/quasar/destination/azure/AzureConfigSpec.scala]: Absolutely. I also found Microsoft offers a [local emulator](https://hub.docker.com/_/microsoft-azure-storage-azurite). I spent some time this morning trying to get it working but couldn't. We could use it in the future to improve testing of all Azure.

Meanwhile I'll add the tests you request 👍

Comment (rintcius) [reply core/src/test/scala/quasar/destination/azure/AzureConfigSpec.scala]: FYI there's a ticket for azurite: ch4057

**Pull request merged**